src/controllers/CurrencyController.ts

```diff
@@ -0,0 +1,49 @@
+import * as RD from '@devexperts/remote-data-ts';
+import * as TE from 'fp-ts/TaskEither';
+import * as E from 'fp-ts/Either';
+import { pipe } from 'fp-ts/function';
+import type { Dispatch, SetStateAction } from "react";
+
+import CurrencyService, { ConversionRatioResponse } from "~/services/currency";
+
+type ControllerDeps = {
+  fetcher: (currency: string) => TE.TaskEither<Error, ConversionRatioResponse>,
+}
+type RatioResponse = { currency: string, ratio: number };
+export type RDRatioResponse = RD.RemoteData<Error, RatioResponse>;
+
+export async function getStaticProps() {
+	const currencyRes = await CurrencyService.fetchCurrenciesListByRelevance();
```

Comment:
You can use `pipe`, `TaskEither` and `Do` notation if needed

Reply:
Didn´t understood how to apply Do notation in this case, so partially solved at [02433a7](https://github.com/rafaelpradella/pradella-fp-page/pull/2/commits/02433a71fa1bcddd215c6281336717063b778513)